### PR TITLE
First pass at documenting Datastore limits

### PR DESCRIPTION
**Pull request by jskeet**
This is to address #903, but there are pending questions about
whether we should actually be batching within the library.
(We'll need to check this thoroughly, in terms of hitting every
overload of every relevant method...)

Not ready for submission, but creating the PR so that it's easily viewable.

### CODE REVIEW

**Comment by jskeet** [comment]
@cc ILMTitan for first review if possible. I'll have another pass at it myself...

File: apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDb.cs [hunk 1]
```diff
@@ -274,7 +282,12 @@ public virtual IReadOnlyList<Key> AllocateIds(IEnumerable<Key> keys, CallSetting
         /// <summary>
         /// Allocates an ID for a single incomplete key asynchronously.
         /// </summary>
-        /// <remarks>This method simply delegates to <see cref="AllocateIdsAsync(IEnumerable{Key},CallSettings)"/>.</remarks>
+        /// <remarks>
+        /// <para>This method simply delegates to <see cref="AllocateIdsAsync(IEnumerable{Key},CallSettings)"/>.</para>
+        /// <para>Datastore limits the number of keys that can be allocated in a single operation. When allocating a large
```
Comment (ILMTitan) [inline apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDb.cs]: I don't think we need this comment here on the single Key method.

File: apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDb.cs [hunk 2]
```diff
@@ -566,6 +611,10 @@ public virtual Task<IReadOnlyList<Key>> InsertAsync(IEnumerable<Entity> entities
         /// <remarks>
         /// <para>On success, if entity has an incomplete key it will be updated in memory with the server-allocated keys.</para>
         /// <para>This method simply delegates to <see cref="Upsert(IEnumerable{Entity}, CallSettings)"/>.</para>
+        /// <para>Datastore limits the number of entities that can be modified in a Commit operation. Although this method does not
+        /// use an existing transaction, it still performs all the work in a single Commit operation. When modifying a large
+        /// number of entities, partition the changes into batches. See
+        /// [Datastore limits](https://cloud.google.com/datastore/docs/concepts/limits) for more details on Datastore limits.</para>
```
Comment (ILMTitan) [inline apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDb.cs]: Don't need this in the single upsert method.

File: apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDb.cs (hunk 1 above)
Comment (jskeet) [inline apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDb.cs]: Done

File: apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDb.cs (hunk 2 above)
Comment (jskeet) [inline apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreDb.cs]: Done.

**Comment by jskeet** [comment]
Transaction methods: I've only got them on Lookup, because that's a separate set of limits to the Commit ones. Have added to Commit, as well as DatastoreDb.Update/InsertAsync.

Thanks for checking - will get Chris to have a look too :)

**Pull request merged**